tests/spec/s-object.js

```diff
@@ -232,6 +234,518 @@ describe('Object', function () {
+
+            obj = Object.defineProperty([], 0, {
+                value: undefined,
+                enumerable: true,
+                writable: true,
+                configurable: true
+            });
+            expect(obj.length).toBe(1);
+            expect(obj[0]).toBe(undefined);
+
+            obj = Object.defineProperty([], 0, {
+                value: null,
+                enumerable: true,
+                writable: true,
+                configurable: true
+            });
+            expect(Object.prototype.hasOwnProperty.call(obj, 0)).toBe(true);
+            expect(obj.length).toBe(1);
+            expect(obj[0]).toBe(null);
+
+            obj = Object.defineProperty([], 1, {
+                enumerable: true,
+                writable: true,
+                configurable: true
+            });
+            expect(obj.length).toBe(2);
+            expect(obj[1]).toBe(undefined);
+
+            obj = Object.defineProperty([], 1, {
+                value: undefined,
+                enumerable: true,
+                writable: true,
+                configurable: true
+            });
+            expect(obj.length).toBe(2);
+            expect(obj[1]).toBe(undefined);
+
+            obj = Object.defineProperty([], 1, {
+                value: null,
+                enumerable: true,
+                writable: true,
+                configurable: true
+            });
+            expect(Object.prototype.hasOwnProperty.call(obj, 1)).toBe(true);
+            expect(obj.length).toBe(2);
+            expect(obj[1]).toBe(null);
+        });
+
+        it('should not throw an error definining elements on arrays using float numbers', function () {
+            var obj = Object.defineProperty([], 1.1, {
+                enumerable: true,
+                writable: true,
+                configurable: true
+            });
+            expect(Object.prototype.hasOwnProperty.call(obj, 1.1)).toBe(true);
+            expect(obj.length).toBe(0);
+            expect(obj[1.1]).toBe(undefined);
+
+            obj = Object.defineProperty([], 1.1, {
+                value: undefined,
+                enumerable: true,
+                writable: true,
+                configurable: true
+            });
+            expect(Object.prototype.hasOwnProperty.call(obj, 1.1)).toBe(true);
+            expect(obj.length).toBe(0);
+            expect(obj[1.1]).toBe(undefined);
+
+            obj = Object.defineProperty([], 1.1, {
+                value: null,
+                enumerable: true,
+                writable: true,
+                configurable: true
+            });
+            expect(Object.prototype.hasOwnProperty.call(obj, 1.1)).toBe(true);
+            expect(obj.length).toBe(0);
+            expect(obj[1.1]).toBe(null);
+        });
+
+        it('should not throw an error definining elements on arrays using trailing point numbers', function () {
+            var obj = Object.defineProperty([], 1., {
```

Comment:
Probably none, I blindly ported everything and as I mentioned, some of this needs thinning out. A lot of my tests are `if you don't try it then you will never know`, but that's how I manage to find some of this stuff. ;)